design/ceph/object/store.md

```diff
@@ -323,31 +323,113 @@ spec:
+found in [Ceph Documentation](https://docs.ceph.com/en/latest/radosgw/s3/commons/#bucket-and-host-name).
+This is supported from Ceph Reef release(v18.0) onwards.
 
-When `rgw_dns_name` is changed for an RGW cluster, all RGWs need to be restarted. To enforce this in Rook, we can apply the `--rgw-dns-name` flag, which will restart RGWs with no user action needed. This is supported from Ceph Reef release(v18.0) onwards.
+Wildcard DNS addressing can be configured in myriad ways. Some options:
+- Kubernetes [ingress loadbalancer](https://kubernetes.io/docs/concepts/services-networking/ingress/#hostname-wildcards)
+- Openshift [DNS operator](https://docs.openshift.com/container-platform/latest/networking/dns-operator.html)
 
-This is different from `customEndpoints` which is used for configuring the object store to replicate and sync data amongst Multisite.
+Rook will implement the following API to allow for user configuration, with detailed design notes to follow.
 
-The default service endpoint for the object store `rook-ceph-rgw-my-store.rook-ceph.svc` and `customEndpoints` in `CephObjectZone` need to be added automatically by the Rook operator otherwise existing object store may impacted. Also check for deduplication in the `rgw_dns_name` list if user manually add the default service endpoint.
+- `hosting` (optional) - configures hosting settings for the CephObjectStore
+    - `advertiseEndpoint` (required) - when hosting settings are configured, users must specify an
+      advertise endpoint so that Rook knows definitively which endpoint the user wants applications
+      to use by default for internal S3 connections.
+    - `dnsNames` (optional) - RGW will reject any S3 connections to unknown endpoints. Users should
+      add any additional endpoints RGW should accept here.
 
-For accessing the bucket point user need to configure wildcard dns in the cluster using [ingress loadbalancer](https://kubernetes.io/docs/concepts/services-networking/ingress/#hostname-wildcards) or in openshift cluster use [dns operator](https://docs.openshift.com/container-platform/latest/networking/dns-operator.html). Same for TLS certificate, user need to configure the TLS certificate for the wildcard dns for the RGW endpoint. This option won't be enabled by default, user need to enable it by adding the `hosting` section in the `Gateway` settings:
+The below example illustrates a commonly-anticipated configuration, and one that Rook can recommend
+to users. The user is using an internal, wildcard-enabled K8s service as the advertised endpoint for
+OBCs. The user also has externally-available ingress that the RGW needs to accept connections from,
+which serves S3 applications outside the Kubernetes cluster. Both endpoints are wildcard-enabled,
+but the internal service is preferable to the external ingress to prevent the ingress router from
+being a cluster-wide S3 bottleneck for Kubernetes applications.
 
 ```yaml
 spec:
   hosting:
+    advertiseEndpoint:
```

Comment:
I'm open to other naming suggestions than `advertiseEndpoint`, though I think this gets the point across pretty well.

I considered `wildcardEndpoint` as an option, but there is no actual guarantee or requirement that this endpoint be wildcarded if the user doesn't want it to be. I think the concept of an advertise endpoint is still useful outside of wildcarding because it would allow users to specifically tell Rook which endpoint to give to end users on OBC/CephObjectStoreUser configmaps.

Reply:
I'd vote for the `advertiseEndpoint` as well